src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1317,9 +1347,7 @@ private function hasReference($id, array $arguments, $deep = false, array &$visi
                         continue;
                     }
 
-                    $arguments = array_merge($service->getMethodCalls(), $service->getArguments(), $service->getProperties());
-
-                    if ($this->hasReference($id, $arguments, $deep, $visited)) {
+                    if ($this->hasReference($id, $service->getMethodCalls(), $deep, $visited) || $this->hasReference($id, $service->getArguments(), $deep, $visited) || $this->hasReference($id, $service->getProperties(), $deep, $visited)) {
```

Comment:
This diff is pure cs, yet is allows to highlight that there is no use of `getOverriddenGetters` here. The reason is that overridden getters are lazy edges, thus should be considered the same as lazy service a few line above.